java.lang.ClassCastException: org.openqa.selenium.remote.RemoteWebElement cannot be cast to io.appium.java_client.MobileElement

**Issue Description along with Code:**

protected WindowsDriver<MobileElement> driver;

java.lang.ClassCastException Issue if directly access the method like below:
System.out.print("Title Check: "+driver.findElementByAccessibilityId("Title).getText());		 

WorkAround:  No java.lang.ClassCastException Issue if cast to the WebElement like below.
WebElement title = driver.findElementByAccessibilityId("Title"); 
getPrintHelper().print("Check the App Title: "+title.getText());

**Environment:**
Maven POM File:
<appium.version>5.0.4</appium.version>
<selenium.version>3.6.0</selenium.version> 
<testng.version>6.11</testng.version>    
   
<dependency>
		<groupId>io.appium</groupId>
		<artifactId>java-client</artifactId>
		<version>${appium.version}</version>
</dependency>
	
<dependency>
    	<groupId>org.seleniumhq.selenium</groupId>
    	<artifactId>selenium-java</artifactId>
    	<version>${selenium.version}</version>
</dependency>

<dependency>
	<groupId>org.testng</groupId>
	<artifactId>testng</artifactId>
	<version>${testng.version}</version>
</dependency>	

Appium server version or git revision: WinAppDriver V1.0 release (File Version 1.0.1710.17001)

Desktop OS/version used to run Appium if necessary: Windows 10
Mobile platform/version under test:  Windows 10 App

**Note:** Looks like same/similar issue as following:
https://github.com/Microsoft/WinAppDriver/issues/286

**Exception stacktraces:** 
https://gist.github.com/aeroclee/b1d3bba03c3188d575b251b557c13cec

Even I'm facing the same issue i.e  **io.appium.java_client.android.AndroidDriver cannot be cast to org.openqa.selenium.interactions.HasTouchScreen**

capabilities.setCapability(MobileCapabilityType.VERSION, Config.get("MobileCapabilityType.VERSION"));
capabilities.setCapability("appPackage", Config.get("appPackage"));
capabilities.setCapability("appActivity", Config.get("appActivity"));
capabilities.setCapability(MobileCapabilityType.APP, Config.get("apkPath"));
URL url = new URL("http://127.0.0.1:4723/wd/hub");
mdriver = new AndroidDriver<MobileElement>(url, capabilities);
			

public static void singleTap(AppiumDriver<MobileElement> mdriver, By locator) {
		try {
			TouchActions action = new TouchActions(mdriver);
			action.singleTap(mdriver.findElement(locator));
			action.perform();
			Assert.assertTrue(true);
		} catch (Exception e) {
			Assert.fail("The  element : " + locator + " is  not visibile/not enabled - '"
					+ "' and the error message is " + e.getMessage());
		}
	}

Mobile.singleTap(mdriver, By.id("io.selendroid.testapp:id/buttonTest"));


Error : java.lang.AssertionError: The  element : By.id: io.selendroid.testapp:id/buttonTest is  not visibile/not enabled  and the error message is io.appium.java_client.android.AndroidDriver cannot be cast to org.openqa.selenium.interactions.HasTouchScreen.

Please help me.

Dear Sir:
I also have the same problem.
I can not use AndroidDriver.
I can only use RemoteWebDriver.
Code:
import java.net.MalformedURLException;
import java.net.URL;
import java.util.concurrent.TimeUnit;

import org.openqa.selenium.WebElement;
import org.openqa.selenium.remote.DesiredCapabilities;
import org.openqa.selenium.remote.RemoteWebDriver;

import io.appium.java_client.MobileBy;
import io.appium.java_client.MobileElement;
import io.appium.java_client.android.AndroidDriver;

public class ScorllBar {
	

	private static RemoteWebDriver driver =null;
	private static AndroidDriver androidDriver=null;

		public static void main(String[] args) throws MalformedURLException, Exception  {
		// TODO Auto-generated method stub
		DesiredCapabilities	cap	=	new	DesiredCapabilities();
		cap.setCapability("platformName", "Android");
		cap.setCapability("platformVersion", 5); 
		cap.setCapability("deviceName", "Android Emulator");
		cap.setCapability("automationName", "Appium");  
		cap.setCapability("appPackage", "tw.com.wgh3h");  
		cap.setCapability("appActivity", "tw.com.demo1.MyMainPage");  
		//RemoteWebDriver driver = new RemoteWebDriver(new URL("http://127.0.0.1:4723/wd/hub"), cap);
		AndroidDriver androidDriver=new AndroidDriver(new URL("http://127.0.0.1:4723/wd/hub"),cap);
		driver = androidDriver;
		driver.manage().timeouts().implicitlyWait(15, TimeUnit.SECONDS);
		
ErrorMessage:
Exception in thread "main" java.lang.NoClassDefFoundError: javax/websocket/DeploymentException
	at io.appium.java_client.android.ListensToLogcatMessages.<clinit>(ListensToLogcatMessages.java:34)
	at ScorllBar.main(ScorllBar.java:32)
Caused by: java.lang.ClassNotFoundException: javax.websocket.DeploymentException
	at java.net.URLClassLoader.findClass(Unknown Source)
	at java.lang.ClassLoader.loadClass(Unknown Source)
	at sun.misc.Launcher$AppClassLoader.loadClass(Unknown Source)
	at java.lang.ClassLoader.loadClass(Unknown Source)
	... 2 more
Test environment:
Java Client:java-client-6.0.0
Appium:1.8.1
Eclipse:Oxygen.3a Release (4.7.3a)

how to switch native to webview am not able to do in iOS app, getting below error whenever i tried using above code
@ronjonesjr @dpgraham @sbonebrake @mykola-mokhnach @imurchie please help

------------------
AppiumDriver
 at io.appium.java_client.AppiumDriver.getContextHandles(AppiumDriver.java:227)
 at ZDebugTests.iOSTest.main(iOSTest.java:28)
Caused by: java.lang.ClassCastException: com.google.common.collect.Maps$TransformedEntriesMap cannot be cast to java.util.List
 at io.appium.java_client.AppiumDriver.getContextHandles﻿

--- a/src/main/java/io/appium/java_client/internal/ElementMap.java
+++ b/src/main/java/io/appium/java_client/internal/ElementMap.java
@@ -37,7 +37,7 @@ public enum ElementMap {
     IOS_XCUI_TEST(AutomationName.IOS_XCUI_TEST.toLowerCase(), IOSElement.class),
     ANDROID_UI_AUTOMATOR(MobilePlatform.ANDROID.toLowerCase(), AndroidElement.class),
     IOS_UI_AUTOMATION(MobilePlatform.IOS.toLowerCase(), IOSElement.class),
-    WINDOWS(MobilePlatform.WINDOWS, WindowsElement.class);
+    WINDOWS(MobilePlatform.WINDOWS.toLowerCase(), WindowsElement.class);
 
 
     private static final Map<String, ElementMap> mobileElementMap;

